packages/internal/src/Polymorphic/Polymorphic.tsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable react/display-name */
+/**
+ * Based on https://blog.logrocket.com/build-strongly-typed-polymorphic-components-react-typescript/
+ */
+
+import React from 'react';
+
+/** An interface that defines the `as` prop */
+export interface AsProp<T extends React.ElementType> {
```

Comment:
let's follow the new directory structure conventions with this new component
`/Polymorphic/Polymorphic.tsx`,  `/Polymorphic/Polymorphic.types.ts`, `/Polymorphic/Polymorphic.styles.ts`